lib/manager.js

```diff
@@ -0,0 +1,106 @@
+"use strict";
+
+let Manifest = require("./manifest");
+let { createFile, generateFingerprint, uriJoin } = require("faucet-pipeline-util");
+let path = require("path");
+
+module.exports = class AssetManager {
+	constructor(referenceDir, { manifestConfig, fingerprint, exitOnError } = {}) {
+		this.referenceDir = referenceDir;
+		this.manifestConfig = manifestConfig;
+		this.fingerprint = fingerprint;
+		this.exitOnError = exitOnError;
+		this.manifest = new Manifest();
+
+		// bind methods for convenience
+		this.writeFile = this.writeFile.bind(this);
+		this.resolvePath = this.resolvePath.bind(this);
+	}
+
+	writeFile(filepath, data, { error } = {}) {
+		let originalPath = filepath;
+		if(this.fingerprint) {
+			filepath = generateFingerprint(filepath, data);
+		}
+
+		return createFile(filepath, data).
+			then(_ => this._updateManifest(originalPath, filepath)).
+			then(_ => {
+				this._report(originalPath, error);
+				if(error && this.exitOnError) {
+					abort();
+				}
+			}).
+			catch(abort);
+	}
+
+	resolvePath(filepath, { enforceRelative } = {}) {
```

Comment:
I'm not sure why this is in here. It seems to have no connection to anything, except that it knows the `referenceDir`. Maybe this is a `util`?

Reply:
Ok, cool